sys/src/loader_interfaces.rs

```diff
@@ -0,0 +1,153 @@
+pub type FnCreateApiLayerInstance = unsafe extern "system" fn(
+    info: *const InstanceCreateInfo,
+    api_layer_info: *const ApiLayerCreateInfo,
+    instance: *mut Instance,
+) -> Result;
+
+/// Loader/API Layer Interface versions
+///
+///  1 - First version, introduces negotiation structure and functions
+pub const CURRENT_LOADER_API_LAYER_VERSION: u32 = 1;
+
+/// Loader/Runtime Interface versions
+/// 
+///  1 - First version, introduces negotiation structure and functions
+pub const CURRENT_LOADER_RUNTIME_VERSION: u32 = 1;
+
+#[repr(transparent)]
+#[derive(Copy, Clone, Eq, PartialEq)]
+pub struct LoaderInterfaceStructureType(i32);
+impl LoaderInterfaceStructureType {
+    pub const UNINTIALIZED: Self = Self(0);
+    pub const LOADER_INFO: Self = Self(1);
+    pub const API_LAYER_REQUEST: Self = Self(2);
+    pub const RUNTIME_REQUEST: Self = Self(3);
+    pub const API_LAYER_CREATE_INFO: Self = Self(4);
+    pub const API_LAYER_NEXT_INFO: Self = Self(5);
+}
+
+impl core::fmt::Debug for LoaderInterfaceStructureType {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let name = match *self {
+            Self::UNINTIALIZED => Some("UNINTIALIZED"),
+            Self::LOADER_INFO => Some("LOADER_INFO"),
+            Self::API_LAYER_REQUEST => Some("API_LAYER_REQUEST"),
+            Self::RUNTIME_REQUEST => Some("RUNTIME_REQUEST"),
+            Self::API_LAYER_CREATE_INFO => Some("API_LAYER_CREATE_INFO"),
+            Self::API_LAYER_NEXT_INFO => Some("API_LAYER_NEXT_INFO"),
+            _ => None,
+        };
+        fmt_enum(fmt, self.0, name)
+    }
+}
+
+#[repr(C)]
+#[derive(Copy, Clone, Debug)]
+pub struct XrNegotiateLoaderInfo {
+    pub ty: LoaderInterfaceStructureType,
+    pub struct_version: u32,
+    pub struct_size: usize,
+    pub min_interface_version: u32,
+    pub max_interface_version: u32,
+    pub min_api_version: Version,
+    pub max_api_version: Version,
+}
+impl XrNegotiateLoaderInfo {
+    pub const TYPE: LoaderInterfaceStructureType = LoaderInterfaceStructureType::LOADER_INFO;
+    pub const VERSION: u32 = 1;
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct XrNegotiateApiLayerRequest {
+    pub ty: LoaderInterfaceStructureType,
+    pub struct_version: u32,
+    pub struct_size: usize,
+    pub layer_interface_version: u32,
+    pub layer_api_version: Version,
+    pub get_instance_proc_addr: Option<pfn::GetInstanceProcAddr>,
+    pub create_api_layer_instance: Option<FnCreateApiLayerInstance>,
+}
+impl XrNegotiateApiLayerRequest {
+    pub const TYPE: LoaderInterfaceStructureType = LoaderInterfaceStructureType::API_LAYER_REQUEST;
+    pub const VERSION: u32 = 1;
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct XrNegotiateRuntimeRequest {
+    pub ty: LoaderInterfaceStructureType,
+    pub struct_version: u32,
+    pub struct_size: usize,
+    /// CURRENT_LOADER_RUNTIME_VERSION
```

Comment:
What's this comment meant to communicate?

Reply:
Its more or less directly copied from the header file, it's meant to communicate that the runtime should set `runtime_interface_version` to `CURRENT_LOADER_RUNTIME_VERSION`.
I overlooked it because it's not directly involved in layer negotiation but if you want I can go over the file again and make the docs more self explanatory.

Reply:
Let's either omit it or make it useful; your call. Generally I'm fine with pointing people at upstream docs, and that goes double for niche advanced use cases like layer authoring.

Reply:
Got it. I'll just remove it for now with the possibility of of adding more detailed documentation in future.